File: uva online judge/11933 - Splitting Numbers.c
#include <stdio.h>

int main(){
    int n, a, b, order, pos;
    while(scanf("%d", &n) && n > 0){
        a = b = order = pos = 0;
        while(n >> pos){
            if(n & (1 << pos)){
                a |= (order ^ 1) << pos;
                b |= (order & 1) << pos;
                order ^= 1;
            }            
            pos += 1;
        }
        printf("%d %d\n", a, b);
    }
    return 0;
}

File: uvaonlinejudge/11933 - Splitting Numbers.py
if __name__ == '__main__':
    while True:
        n = int(input())
        if n == 0:
            break
        a, b, order, pos = 0, 0, 0, 0
        while n >> pos > 0:
            if n & (1 << pos) > 0:
                a |= (order ^ 1) << pos
                b |= (order & 1) << pos
                order ^= 1
            pos += 1
        print(a, b)
